design/role.go

```diff
@@ -70,6 +71,21 @@ var _ = a.Resource("resource_roles", func() {
 		a.Response(d.NotFound, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+	a.Action("assignRole", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.PATCH("/:resourceID/roles/:roleName"),
+		)
+		a.Payload(updateUserIDList) // should refactor this variable's name in collaborators design definition too.
```

Comment:
I'm wondering if the payload structure is ideal.  Will the `"type"` ever be any value other than `"identities"` ?  Also, thinking about the workflow, I can understand how this endpoint is useful for "batch" assigning a single role to a number of users, however how about assigning multiple roles to a single user, as is possible in the current wireframes that we've been reviewing from the UX team?  Will this be done as multiple calls to the auth service? If so, that seems inefficient.

Reply:
To add to my previous comment, this is the wireframe for what I'm referring to:

[https://redhat.invisionapp.com/share/QNGBDSOHP6K#/screens/294869601](https://redhat.invisionapp.com/share/QNGBDSOHP6K#/screens/294869601)

In the UI shown here, it is possible to select multiple roles to assign to a single user.  I am certain that we don't wish this operation to require multiple calls to the auth service.

Reply:
> I'm wondering if the payload structure is ideal. Will the "type" ever be any value other than "identities"

For now no. But we can remove the enum, and/or add other constants to the enum as well. I was re-using the existing payload at the moment.

> I am certain that we don't wish this operation to require multiple calls to the auth service.

I don't have a good answer for this honestly - and might go for multiple calls to the auth service for now since we wouldn't be having too many roles to start with. So for assigning 3 roles to 50 users, there's gonna be 3 requests which IMO isn't gonna harm the user experience, if they are all async. 

The alternative would be to include the rolename in the payload itself. Wouldn't that make the transaction too big? Or should we have multiple transactions for each payload item?